src/cmd/cloud.js

```diff
@@ -41,311 +43,381 @@ const PLATFORMS = extend(utilities.knownPlatforms(), {
+				return createAPI().flashDevice(deviceId, fileMapping, targetVersion, product);
+			})
+			.then((resp) => {
+				if (resp.status || resp.message){
+					this.ui.stdout.write(`Flash device OK: ${resp.status || resp.message}${os.EOL}`);
+				} else if (resp.output === 'Compiler timed out or encountered an error'){
+					this.ui.stdout.write(`${os.EOL}${(resp.errors && resp.errors[0])}${os.EOL}`);
+					throw new Error('Compiler encountered an error');
+				} else {
+					throw normalizedApiError(resp);
+				}
+			})
+			.then(() => {
+				if (!product){
+					return;
+				}
+				[
+					`device ${deviceId} is now marked as a developement device and will NOT receive automatic product firmware updates.`,
+					'to resume normal updates, please visit:',
+					`https://console.particle.io/${product}/devices/unmark-development/${deviceId}`
+				].forEach(line => this.ui.stdout.write(`${line}${os.EOL}`));
+			})
+			.catch(err => {
+				throw normalizedApiError(err);
+			});
 	}
 
-	_flashKnownApp({ api, deviceId, filePath }) {
-		if (!settings.knownApps[filePath]) {
+	_flashKnownApp({ product, deviceId, filePath }){
+		if (!settings.knownApps[filePath]){
 			throw new VError(`I couldn't find that file: ${filePath}`);
 		}
 
-		return api.getAttributes(deviceId).then((attrs) => {
-			const spec = _.find(specs, { productId: attrs.product_id });
-			if (spec) {
-				if (spec.knownApps[filePath]) {
-					return api._populateFileMapping( { list: [spec.knownApps[filePath]] } );
-				}
+		return createAPI().getDeviceAttributes(deviceId)
+			.then((attrs) => {
+				const spec = _.find(specs, { productId: attrs.product_id });
+
+				if (spec){
+					if (spec.knownApps[filePath]){
+						return populateFileMapping({ list: [spec.knownApps[filePath]] });
+					}
 
-				if (spec.productName) {
-					throw new VError(`I don't have a ${filePath} binary for ${spec.productName}.`);
+					if (spec.productName){
+						throw new VError(`I don't have a ${filePath} binary for ${spec.productName}.`);
+					}
 				}
-			}
 
-			// TODO: this shouldn't be necessary. Just look at the attrs.platform_id instead of
-			// product_id above
-			return prompt([{
-				name: 'type',
-				type: 'list',
-				message: 'Which type of device?',
-				choices: [
-					'Photon',
-					'Core',
-					'P1',
-					'Electron'
-				]
-			}]).then((ans) => {
-				const spec = _.find(specs, { productName: ans.type });
-				const binary = spec && spec.knownApps[filePath];
-
-				if (!binary) {
-					throw new VError(`I don't have a ${filePath} binary for ${ans.type}.`);
-				}
-
-				return { map: { binary: binary } };
+				// TODO (mirande): fix this
+				// TODO: this shouldn't be necessary. Just look at the attrs.platform_id instead of
+				// product_id above
+				const question = {
+					name: 'type',
+					type: 'list',
+					message: 'Which type of device?',
+					choices: ['Photon', 'Core', 'P1', 'Electron']
+				};
+
+				return prompt(question)
```

Comment:
Remove this question. It's not necessary.

Reply:
✅ in https://github.com/particle-iot/particle-cli/pull/565/commits/934d2e14bf3ce2ee5cf1d295117107fc939be7e0